js/tji-datasets-explore.js

```diff
@@ -1,15 +1,349 @@
+    keys.push(this.missing_data_label);
+  }
+  var counts = _.map(keys, function(k){ return grouped[k].length});
+  return {
+    keys: keys,
+    counts: counts
+  };
+}
+
+
+// ********************************************************************
+// * Extends TJIGroupByBarChart... but for doughnuts.
+// ********************************************************************
+
+
+var TJIGroupByDoughnutChart = function(elt_id, groupBy, data, missing_data_label) {
+  TJIGroupByBarChart.call(this, elt_id, groupBy, data, missing_data_label);
+}
+TJIGroupByDoughnutChart.prototype = Object.create(TJIGroupByBarChart.prototype);
+TJIGroupByDoughnutChart.prototype.constructor = TJIGroupByDoughnutChart;
+TJIGroupByDoughnutChart.prototype.type = 'doughnut';
+TJIGroupByDoughnutChart.prototype.color_palette = [
+  COLOR_TJI_BLUE, COLOR_TJI_RED, COLOR_TJI_DEEPBLUE, COLOR_TJI_PURPLE,
+  COLOR_TJI_YELLOW, COLOR_TJI_TEAL, COLOR_TJI_DEEPRED, COLOR_TJI_DEEPPURPLE,
+];
+TJIGroupByDoughnutChart.prototype.get_options_overrides = function() {
+  return {
+    scales: {},
+    legend: {
+      display: true,
+      position: 'bottom',
+      labels: {
+        fontSize: 12,
+      }
+    },
+    pieceLabel: {
+      mode: function (args) {
+        return args.percentage + '%';
+      },
+      precision: 0,
+      showZero: true,
+      fontSize: 10,
+      fontColor: '#fff',
+      // available value is 'default', 'border' and 'outside'
+      position: 'default'
+    }
+  };
+}
+
+
+// ********************************************************************
+// * "Class" that creates and manages all charts and the filter panel.
+// *
+// * Constructor arguments:
+// *   chart_configs: array of chart config objects,
+// *                  e.g. [{type: 'bar', group_by:'year'}, ...]
+// *   charts_elt_id: id of HTML element to put charts in
+// *   filters_elt_id: id of HTML element to put filter checkboxes in
+// *   chart_wrapper: HTML for wrapper around chart canvas object
+// *   count_template: HTML for record-count element, containing "{count}"
+// *                   somewhere (which will be replaced with the record count).
+// ********************************************************************
+
+
+var ChartView = function(chart_configs, charts_elt_id, filters_elt_id, chart_wrapper, count_template){
+
+  this.state = {
+    data: null,
+    active_filters: [], //put active filters here
+    charts: [],
+    $count: null
+  }
+
+  this.chart_configs = chart_configs;
+  this.charts_elt_id = charts_elt_id;
+  this.filters_elt_id = filters_elt_id;
+  this.chart_wrapper = chart_wrapper;
+  this.count_template = count_template;
+  this.filters = null;
+
+  this.get_data();
+}
+
+ChartView.prototype.missing_data_label = '(not given)';
+
+ChartView.prototype.get_data = function() {
+  var that = this;
+  jQuery.getJSON('/cleaned_custodial_death_reports.json')
+    .done(function(data){
+      that.state.data = data;
+      that.parse_data();
+      that.create_charts();
```

Comment:
I think @wetchler, you mentioned that this method is doing a lot more than "getting data".
We could make a new method called ChartView.prototype.get_data_callback and throw this.create_chart, this.create_filter_panel, this.attach_events in there. 🤷‍♂️

Reply:
Say more? I don't quite understand what you have in mind